src/test/java/org/jvnet/hudson/plugins/shelveproject/ShelveProjectActionTest.java

```diff
@@ -21,37 +24,39 @@
 public class ShelveProjectActionTest {
     @Rule
     public JenkinsRule jenkinsRule = new JenkinsRule();
+    private FreeStyleProject project = null;
 
     @Before
-    public void setUp() {
+    public void setUp() throws IOException{
+        project = jenkinsRule.createFreeStyleProject();
         jenkinsRule.jenkins.setSecurityRealm(jenkinsRule.createDummySecurityRealm());
         jenkinsRule.jenkins.setAuthorizationStrategy(new MockAuthorizationStrategy().
-                grant(Jenkins.ADMINISTER).everywhere().to("admin").
-                grant(Jenkins.READ, Item.DELETE).everywhere().to("developer").
-                grant(Jenkins.READ, Item.CREATE).everywhere().to("creator").
-                grant(Jenkins.READ).everywhere().to("reader"));
+                grant(Jenkins.ADMINISTER).onItems(project).to("admin").
+                grant(Jenkins.READ, Item.DELETE).onItems(project).to("developer").
+                grant(Jenkins.READ, Item.CREATE).onItems(project).to("creator").
+                grant(Jenkins.READ).onItems(project).to("reader"));
     }
 
-    @Issue("JENKINS-55462")
+    @Issue({"JENKINS-55462", "JENKINS-66382"})
```

Comment:
This test is not covering JENKINS-66382:

```suggestion
    @Issue("JENKINS-55462")
```